### PR TITLE
Updates Gnosis Form

### LINKED ISSUE #426: Update Gnosis Config Form
- [ ] Remove Add address
- [ ] Add Number of Signers Input field, (set up to control the number of address line)

This is to improve UX by feeling more like a n of m schema

### PR DESCRIPTION
**Pull request by Da-Colon**
## Overview

Small Update to the UI/UX of the Gnosis Config Form. Instead of click an 'add' button, the user enters a number in the field

## Screenshot
image

closes #426

### CODE REVIEW

**Review by adamgall — CHANGES_REQUESTED** [review]
This code must have been added in the original Gnosis UI PR, but there's a typo in the error displayed when Signature Threshold is higher than signers:

"Threshold is to high"

to -> too

edit: this is fixed, thanks

**Review by adamgall — CHANGES_REQUESTED** [review]
You can get into an invalid state ("Deploy" button enabled), when entering a number into "Signers" (like 3), then deleting it (which turns it back into 0), then filling out addresses in the three address inputs (which stuck around, despite there being 0 signers).

Screen Shot 2022-08-21 at 12 00 19 AM

**Review by adamgall — CHANGES_REQUESTED** [review]
Bug: can make Signature Threshold value be greater than Signer value, without an error appearing

Screen Shot 2022-08-21 at 12 11 52 AM
 
To trigger bug start with a valid m of n schema, then use the "remove" buttons next to addresses to decrease the Signers to below Signature Threshold.

File: src/components/DaoCreator/gnosis/GnosisConfig.tsx [hunk 1]
```diff
@@ -55,17 +55,20 @@ export function GnosisConfig() {
...
+
+  useEffect(() => {
     let error: string = '';
-    if (Number(threshold) === 0) {
+    if (Number(signatureThreshold) === 0) {
       error = 'Threshold must be greater than 0';
     }
-    if (Number(threshold) > trustedAddresses.length) {
+    if (Number(signatureThreshold) > numberOfSigners) {
       error = 'Threshold is to high';
```
Comment (adamgall) [inline src/components/DaoCreator/gnosis/GnosisConfig.tsx:67]: to -> too

**Review by adamgall — DISMISSED** [review]

**Comment by Da-Colon** [comment]
@adamgall merged w/ develop to fix conflict, will need a re-approval

**Review by adamgall — APPROVED** [review]

**Pull request merged**